beamer/health/notification_state.py

```diff
@@ -0,0 +1,34 @@
+import json
+import os
+from pathlib import Path
+
+
+class NotificationState:
+    cache_path: Path
+
+    def __init__(self, cache_path: Path) -> None:
+        self.cache_path = cache_path
+
+        try:
+            with open(os.path.join(self.cache_path, "breadcrumbs.json"), "r") as f:
+                self.notified_state = json.load(f)
+        except FileNotFoundError:
+            self.notified_state = {}
```

Comment:
Please consider making `notified_state` private if it is not used outside of this class.